lua/apisix/plugins/proxy-cache.lua

```diff
@@ -178,6 +158,91 @@ local function match_method_and_status(conf, ctx)
     return false
 end
 
+-- refer to https://gist.github.com/titpetric/ed6ec548af160e82c650cf39074878fb
+local function file_exists(name)
+    local f = io_open(name, "r")
+    if f~=nil then io_close(f) return true else return false end
+end
+
+
+local function explode(d, p)
+    local t, ll
+    t={}
+    ll=0
+    if(#p == 1) then return {p} end
+    while true do
+        local l=string.find(p, d, ll, true) -- find the next d in the string
+        if l~=nil then -- if "not not" found then..
+                tab_insert(t, string.sub(p, ll, l-1)) -- Save it in our array.
+                ll=l+1 -- save just after where we found it for searching next time.
+        else
+                tab_insert(t, string.sub(p, ll)) -- Save what's left in our array.
+                break -- Break at end, as it should be, according to the lua manual.
+        end
+    end
+    return t
+end
+
+
+local function generate_cache_filename(cache_path, cache_levels, cache_key)
+    local md5sum = ngx.md5(cache_key)
+    local levels = explode(":", cache_levels)
+    local filename = ""
+
+    local index = string.len(md5sum)
+    for k, v in pairs(levels) do
+            local length = tonumber(v)
+            -- add trailing [length] chars to index
+            index = index - length;
+            filename = filename .. md5sum:sub(index+1, index+length) .. "/";
+    end
+    if cache_path:sub(-1) ~= "/" then
+            cache_path = cache_path .. "/";
+    end
+    filename = cache_path .. filename .. md5sum
+    return filename
+end
+
+
+local function cache_purge(conf, ctx)
+    local cache_zone_info = ngx_re.split(ctx.var.upstream_cache_zone_info, ",")
+
+    local filename = generate_cache_filename(cache_zone_info[1], cache_zone_info[2],
+                                             ctx.var.upstream_cache_key)
+    if file_exists(filename) then
+        os.remove(filename)
```

Comment:
Only privileged agents can delete cached files. 
The worker process defaults to the nobody user, which does not have this permission.

Reply:
Don't worry. If the worker has permission to create it, then it has permission to delete it.

Reply:
that is fine.